library/Haste/Model/Relations.php

```diff
@@ -666,6 +738,90 @@ public function addRelationFilters($dc)
         return $return . '</div>';
     }
 
+    /**
+     * Adds search fields for relations.
+     *
+     * @param $dc
+     *
+     * @return string
+     */
+    public function addRelationSearch($dc)
+    {
+        if (empty(static::$arrSearchableFields)) {
+            return '';
+        }
+
+        $return = '<div class="tl_filter tl_subpanel">';
+        $session = \Session::getInstance();
+        $sessionValues = $session->get('haste_search');
+
+        // Search field per relation
+        foreach (static::$arrSearchableFields as $field => $arrRelation) {
+
+            // Get searchable fields from related table
+            $relatedSearchFields = [];
+            $relTable = $arrRelation['related_table'];
+
+            \Controller::loadDataContainer($relTable);
+            foreach ((array) $GLOBALS['TL_DCA'][$relTable]['fields'] as $relatedField => $dca) {
+                if (isset($dca['search']) && true === $dca['search']) {
```

Comment:
I wouldn't bother here that much and just use `if ($dca['search']) {` or `if (isset($dca['search']) && $dca['search']) {` to avoid unnecessary notices.